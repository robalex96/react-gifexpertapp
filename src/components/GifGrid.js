import React, { Fragment } from 'react';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({category}) => {    
    const {data:gifs, loading} = useFetchGifs(category);

    return (
        <Fragment>
            <h3 className='animate__animated animate__fadeIn'>{category}</h3>
            {loading && <p>Loading...</p>}
            <div className='card-grid'>
                {
                    gifs.map(gif =>
                        (
                        <GifGridItem 
                            key={gif.id}
                            {...gif}
                        />
                        )
                    )
                }
            </div>
        </Fragment>
    )
};
