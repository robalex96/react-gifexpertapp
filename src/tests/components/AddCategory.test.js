import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('AddCategory component', () => {
    const setCategories = jest.fn();
    let wrapper;

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });

    test('Render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Submit should not be excetuted', () => {
        const form = wrapper.find('form');
        form.simulate('submit', {preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('Submit should be excetuted setCategories and input clean', () => {
        const value = 'Prueba';

        const input = wrapper.find('input');
        input.simulate('change', { target: {value}});
        
        const form = wrapper.find('form');
        form.simulate('submit', {preventDefault(){}});
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        expect(input.prop('value')).toBe('');
    });    
});