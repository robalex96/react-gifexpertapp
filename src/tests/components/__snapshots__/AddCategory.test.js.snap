// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory component Render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
